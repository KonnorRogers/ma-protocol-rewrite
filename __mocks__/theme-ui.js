const themeUI = jest.genMockFromModule("theme-ui")

let color = "dark"

function setColorMode(mode) {
  color = mode
}

function useColorMode() {
  return [color, setColorMode]
}

themeUI.setColorMode = setColorMode
themeUI.useColorMode = useColorMode
module.exports = themeUI
