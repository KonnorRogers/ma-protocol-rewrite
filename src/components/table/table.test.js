import * as React from "react"
import { render } from "@testing-library/react"
import { Table, TableCell, TableRow, TableItem } from "./index.js"

// Mocked colorMode hook
import { setColorMode } from "theme-ui"

function renderTable() {
  return render(
    <Table>
      <TableRow heading={true}>
        <TableItem> Hi there </TableItem>
        <TableItem> Hi there </TableItem>
        <TableItem> Hi there </TableItem>
        <TableItem> Hi there </TableItem>
      </TableRow>
      <TableRow>
        <TableItem>Yo dawg</TableItem>
        <TableItem>Yo dawg</TableItem>
        <TableItem>Yo dawg</TableItem>
        <TableItem>Yo dawg</TableItem>
      </TableRow>
      <TableRow>
        <TableItem>
          <TableCell>Table Cell for centering</TableCell>
        </TableItem>
        <TableItem>Testing 123</TableItem>
        <TableItem>Testing 123</TableItem>
        <TableItem>Testing 123</TableItem>
      </TableRow>
    </Table>
  )
}

describe("Table components", () => {
  test("Should properly render a table with darkMode", () => {
    setColorMode("dark")
    const { asFragment } = renderTable()

    expect(asFragment()).toMatchSnapshot()
  })
  test("Should properly render a table with lightMode", () => {
    setColorMode("light")
    const { asFragment } = renderTable()

    expect(asFragment()).toMatchSnapshot()
  })
})
