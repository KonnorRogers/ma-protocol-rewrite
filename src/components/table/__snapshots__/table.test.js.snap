// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table components Should properly render a table with darkMode 1`] = `
<DocumentFragment>
  .emotion-16 {
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 1rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  border: 1px solid white;
  min-width: 500px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-16 div:last-of-type {
  border-bottom: none;
}

.emotion-4 {
  font-weight: bold;
  font-size: 1.15rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  text-align: center;
  width: 100%;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border-bottom: 1px solid white;
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.emotion-4 div:last-of-type {
  border-right: none;
}

.emotion-4 div:last-of-type:after {
  content: "";
  display: block;
  position: absolute;
  right: -2rem;
  width: 2rem;
  height: 1px;
}

.emotion-0 {
  min-width: 0px;
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  border-right: 1px solid white;
  padding: 0.5rem;
  text-align: center;
  position: relative;
}

.emotion-9 {
  font-weight: normal;
  font-size: initial;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  text-align: center;
  width: 100%;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border-bottom: 1px solid white;
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.emotion-9 div:last-of-type {
  border-right: none;
}

.emotion-9 div:last-of-type:after {
  content: "";
  display: block;
  position: absolute;
  right: -2rem;
  width: 2rem;
  height: 1px;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.emotion-16 {
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 1rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  border: 1px solid white;
  min-width: 500px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-16 div:last-of-type {
  border-bottom: none;
}

.emotion-4 {
  font-weight: bold;
  font-size: 1.15rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  text-align: center;
  width: 100%;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border-bottom: 1px solid white;
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.emotion-4 div:last-of-type {
  border-right: none;
}

.emotion-4 div:last-of-type:after {
  content: "";
  display: block;
  position: absolute;
  right: -2rem;
  width: 2rem;
  height: 1px;
}

.emotion-0 {
  min-width: 0px;
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  border-right: 1px solid white;
  padding: 0.5rem;
  text-align: center;
  position: relative;
}

.emotion-9 {
  font-weight: normal;
  font-size: initial;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  text-align: center;
  width: 100%;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border-bottom: 1px solid white;
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.emotion-9 div:last-of-type {
  border-right: none;
}

.emotion-9 div:last-of-type:after {
  content: "";
  display: block;
  position: absolute;
  right: -2rem;
  width: 2rem;
  height: 1px;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

<div
    class="emotion-16"
  >
    <div
      class="emotion-4"
    >
      <div
        class="emotion-0"
      >
         Hi there 
      </div>
      <div
        class="emotion-0"
      >
         Hi there 
      </div>
      <div
        class="emotion-0"
      >
         Hi there 
      </div>
      <div
        class="emotion-0"
      >
         Hi there 
      </div>
    </div>
    <div
      class="emotion-9"
    >
      <div
        class="emotion-0"
      >
        Yo dawg
      </div>
      <div
        class="emotion-0"
      >
        Yo dawg
      </div>
      <div
        class="emotion-0"
      >
        Yo dawg
      </div>
      <div
        class="emotion-0"
      >
        Yo dawg
      </div>
    </div>
    <div
      class="emotion-9"
    >
      <div
        class="emotion-0"
      >
        <div
          class="emotion-10"
        >
          Table Cell for centering
        </div>
      </div>
      <div
        class="emotion-0"
      >
        Testing 123
      </div>
      <div
        class="emotion-0"
      >
        Testing 123
      </div>
      <div
        class="emotion-0"
      >
        Testing 123
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Table components Should properly render a table with lightMode 1`] = `
<DocumentFragment>
  .emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.emotion-16 {
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 1rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  border: 1px solid black;
  min-width: 500px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-16 div:last-of-type {
  border-bottom: none;
}

.emotion-4 {
  font-weight: bold;
  font-size: 1.15rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  text-align: center;
  width: 100%;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border-bottom: 1px solid black;
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.emotion-4 div:last-of-type {
  border-right: none;
}

.emotion-4 div:last-of-type:after {
  content: "";
  display: block;
  position: absolute;
  right: -2rem;
  width: 2rem;
  height: 1px;
}

.emotion-0 {
  min-width: 0px;
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  border-right: 1px solid black;
  padding: 0.5rem;
  text-align: center;
  position: relative;
}

.emotion-9 {
  font-weight: normal;
  font-size: initial;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  text-align: center;
  width: 100%;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border-bottom: 1px solid black;
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.emotion-9 div:last-of-type {
  border-right: none;
}

.emotion-9 div:last-of-type:after {
  content: "";
  display: block;
  position: absolute;
  right: -2rem;
  width: 2rem;
  height: 1px;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.emotion-16 {
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 1rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  border: 1px solid black;
  min-width: 500px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-16 div:last-of-type {
  border-bottom: none;
}

.emotion-4 {
  font-weight: bold;
  font-size: 1.15rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  text-align: center;
  width: 100%;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border-bottom: 1px solid black;
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.emotion-4 div:last-of-type {
  border-right: none;
}

.emotion-4 div:last-of-type:after {
  content: "";
  display: block;
  position: absolute;
  right: -2rem;
  width: 2rem;
  height: 1px;
}

.emotion-0 {
  min-width: 0px;
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  border-right: 1px solid black;
  padding: 0.5rem;
  text-align: center;
  position: relative;
}

.emotion-9 {
  font-weight: normal;
  font-size: initial;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  text-align: center;
  width: 100%;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border-bottom: 1px solid black;
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.emotion-9 div:last-of-type {
  border-right: none;
}

.emotion-9 div:last-of-type:after {
  content: "";
  display: block;
  position: absolute;
  right: -2rem;
  width: 2rem;
  height: 1px;
}

<div
    class="emotion-16"
  >
    <div
      class="emotion-4"
    >
      <div
        class="emotion-0"
      >
         Hi there 
      </div>
      <div
        class="emotion-0"
      >
         Hi there 
      </div>
      <div
        class="emotion-0"
      >
         Hi there 
      </div>
      <div
        class="emotion-0"
      >
         Hi there 
      </div>
    </div>
    <div
      class="emotion-9"
    >
      <div
        class="emotion-0"
      >
        Yo dawg
      </div>
      <div
        class="emotion-0"
      >
        Yo dawg
      </div>
      <div
        class="emotion-0"
      >
        Yo dawg
      </div>
      <div
        class="emotion-0"
      >
        Yo dawg
      </div>
    </div>
    <div
      class="emotion-9"
    >
      <div
        class="emotion-0"
      >
        <div
          class="emotion-10"
        >
          Table Cell for centering
        </div>
      </div>
      <div
        class="emotion-0"
      >
        Testing 123
      </div>
      <div
        class="emotion-0"
      >
        Testing 123
      </div>
      <div
        class="emotion-0"
      >
        Testing 123
      </div>
    </div>
  </div>
</DocumentFragment>
`;
